src/main/java/net/minecraftforge/registries/holdersets/CompositeHolderSet.java

```diff
@@ -200,4 +203,40 @@ public List<HolderSet<T>> homogenize()
         }
         return outputs;
     }
+    
+    /**
+     * @return True if all of our sub-holdersets have the same {@link SerializationType} (string, list, or object).
+     * False if we have more than one holderset AND if either we have more than one serialization type among them,
+     * or any holderset is {@link SerializationType.UNKNOWN}.
+     */
+    public boolean isHomogenous()
+    {
+        final List<HolderSet<T>> holderSets = this.getComponents();
+        
+        if (holderSets.size() < 2)
+        {
+            return true;
+        }
+        
+        // Get the first holderset and check if any subsequent holdersets are different
+        SerializationType firstType = holderSets.get(0).serializationType();
+        
+        // If any holderset has an unknown type, then we cannot assume we are homogenous
+        if (firstType == SerializationType.UNKNOWN)
+        {
+            return false;
+        }
+        
+        final int size = holderSets.size();
+        for (int i=1; i<size; i++)
```

Comment:
Formatting: spaces between operators.
```suggestion
        for (int i = 1; i < size; i++)
```